src/Modal/Modal.jsx

```diff
@@ -1,99 +1,132 @@
-                <label htmlFor = "watch_later">Watch later</label>
-            </div>
-            {
-                myPlaylist.myLibrary.map(library=>{
-                    return (
-                        <div className = "library">
-                            <input 
-                            type = "checkbox" 
-                            checked = {library.checked} 
-                            className = "checkbox"
-                            name = {library.name}
-                            onChange = {() => (checkboxHandler(library,video))}/>
-                            <label htmlFor = {library}>{library.name}</label>
-                        </div>
-                    )
-                })
-            }
-        </div>
-        
+  };
+
+  const isVideoInLibrary = (videoId, libraryId) => {
+    const currentLibrary = myPlaylist.myLibrary.find( library => library._id === libraryId);
+    const videoIdsInCurrentLibrary = currentLibrary.videos.map(video => video._id)
+    return videoIdsInCurrentLibrary.includes(videoId)
+  }
+
+  // const handleKeyPress = (e) => {
+  //   if (e.key === "Enter" && newLibrary) {
+  //     createNewLibraryHandler();
+  //   }
+  // };
+  return (
+    <div className="modal">
+      <div className="save flex">
+        <div>Save to...</div>
+        <button className="close-btn" onClick={closeHandler}>
+          <i className="fa fa-close"></i>
+        </button>
+      </div>
+      <div className="library-list">
         {
-            openForm ? <div className = "new_playlist">
-                <input 
-                placeholder = "Enter Playlist Name" 
-                className = "new-playlist__input"
-                value = {newLibrary}
-                onChange = {inputChangeHandler}
-                onKeyPress = {handleKeyPress}
-                />
-                {error && <div className = "error">{error}</div>}
-                <div className = "create-btn-container">
-                    <button className = "form__create-btn" onClick = {createNewLibraryHandler}>CREATE</button>
-                </div>
-                </div>:
-            <div className = "new_playlist">
-                <button className = "create-btn" onClick = {createNewPlaylistHandler}><span className = "plus">+</span><span>Create new playlist</span></button>
+        myPlaylist.myLibrary.length? 
+        myPlaylist.myLibrary.map((library) => {
+          return (
+            <div className="library" key={library._id}>
+              <input
+                type="checkbox"
+                checked={isVideoInLibrary(video._id,library._id)}
+                className="checkbox"
+                name={library.name}
+                onChange={(e) => checkboxHandler(library._id, video._id)}
+              />
+              <label htmlFor={library}>{library.name}</label>
             </div>
-        }
+          );
+        }) : null
+        } 
+      </div>
+
+      {openForm ? (
```

Comment:
Good practice: ternary operator should be used only if code extends till 3 lines maximum, else move it to function for cleaner looking code.

Reply:
done